tests/BindModelTests.js

```diff
@@ -0,0 +1,136 @@
+
+gpii.tests.nexus.bindModel.testDefs = [
+    {
+        name: "Bind Model",
+        gradeNames: "gpii.test.nexus.testCaseHolder",
+        expect: 6,
+        config: {
+            configName: "gpii.tests.nexus.config",
+            configPath: configPath
+        },
+        testComponentPath: "nexusBindModelTestComponent",
+        testModelPath: "someModelPath",
+        components: {
+            client: {
+                type: "gpii.tests.nexus.bindModel.wsClient"
+            }
+        },
+        events: {
+            targetModelChanged: null
+        },
+        /*
+        // TODO: See gpii.tests.nexus.bindModel.registerModelListenerForPath
+        invokers: {
+            fireTargetModelChanged: {
+                "this": "{testCaseHolder}.events.targetModelChanged",
+                method: "fire"
+            }
+        },
+        */
+        sequence: [
+            {
+                func: "gpii.test.nexus.assertNoComponentAtPath",
+                args: ["Component not yet constructed", "{tests}.options.testComponentPath"]
+            },
+            {
+                func: "{constructComponentRequest}.send",
+                args: [gpii.tests.nexus.bindModel.componentOptions]
+            },
+            {
+                event: "{constructComponentRequest}.events.onComplete",
+                listener: "gpii.test.nexus.assertStatusCode",
+                args: ["{constructComponentRequest}", 200]
+            },
+            {
+                func: "gpii.tests.nexus.bindModel.registerModelListenerForPath",
+                args: [
+                    "{tests}.options.testComponentPath",
+                    "{tests}.options.testModelPath",
+                    "{testCaseHolder}.events.targetModelChanged"
+                ]
+            },
+            {
+                func: "{client}.connect"
+            },
+            {
+                event: "{client}.events.onConnect",
+                listener: "fluid.identity"
+            },
+            {
+                event: "{client}.events.onReceiveMessage",
+                listener: "jqUnit.assertEquals",
+                args: ["Received initial message with the state of the component's model", 2, "{arguments}.0"]
+            },
+            {
+                func: "{client}.send",
+                args: [
+                    {
+                        path: "",
+                        value: 10
+                    }
+                ]
+            },
+            // TODO: Can I rely on the targetModelChanged and onReceiveMessage events happening in this order?
+            {
+                event: "{testCaseHolder}.events.targetModelChanged",
+                listener: "gpii.test.nexus.assertComponentModel",
+                args: ["Model updated", "{tests}.options.testComponentPath", { someModelPath: 10 }]
+            },
+            {
+                event: "{client}.events.onReceiveMessage",
+                // TODO: Do I want a message here?
```

Comment:
I think a client should receive notifications of its own changes - it will be avoidable once https://github.com/fluid-project/infusion/pull/671 is merged in but I think we will still want to do it